### PR TITLE
feat(cpu_monitor): Window averaging

### PR DESCRIPTION
**Pull request by reinzor**
Added `~window` parameter in order to average multiple cpu percentage readings.

### CODE REVIEW

**Comment by reinzor** [comment]
Ping @g-gemignani

File: diagnostic_common_diagnostics/src/diagnostic_common_diagnostics/cpu_monitor.py [hunk 1]
```diff
@@ -34,35 +34,45 @@
...
 class CpuTask(DiagnosticTask):
-    def __init__(self, warning_percentage):
+    def __init__(self, warning_percentage, window):
         DiagnosticTask.__init__(self, "CPU Information")
         self._warning_percentage = int(warning_percentage)
+        self._readings = collections.deque(maxlen=window)
+
+    def _get_average_reading(self):
+        def avg(lst):
+            return float(sum(lst)) / len(lst)
```
Comment (g-gemignani) [inline diagnostic_common_diagnostics/src/diagnostic_common_diagnostics/cpu_monitor.py]: Could you please also handle the len(lst) = 0 case just to be on the safe side?

**Review by g-gemignani — CHANGES_REQUESTED** [review]
Would adding a test be too much to ask?

**Comment by g-gemignani** [comment]
On second thought, @reinzor do you need this improvement also in indigo? I would otherwise just add this new feature into noetic

File: diagnostic_common_diagnostics/src/diagnostic_common_diagnostics/cpu_monitor.py (hunk 1 above)
Comment (reinzor) [reply diagnostic_common_diagnostics/src/diagnostic_common_diagnostics/cpu_monitor.py]: Done

**Comment by reinzor** [comment]
> On second thought, @reinzor do you need this improvement also in indigo? I would otherwise just add this new feature into noetic

It would be nice to have this in `Melodic` as well (it is backwards compatible and should not affect any running configuration). Since `Kinetic`/`Melodic` are using the `indigo-devel` branch that would be nice :).

**Comment by reinzor** [comment]
Thanks btw for continuing to remain this repo!

**Comment by reinzor** [comment]
> Would adding a test be too much to ask?

Regarding the test: the CPU monitor is already being tested here: https://github.com/ros/diagnostics/tree/indigo-devel/diagnostic_common_diagnostics/test/launch

**Comment by g-gemignani** [comment]
Correct, but since you are adding a feature I believe we should extend the test :)

**Comment by g-gemignani** [comment]
Yep, it is true that the analyzer is tested there but I was thinking about creating another one where we set the window parameter to something different than 1, but I guess it is not needed.

**Review by g-gemignani — APPROVED** [review]

**Pull request merged**